app/Http/Controllers/Home/BookingController.php

```diff
@@ -115,4 +119,110 @@ public function bedCount($cityId, $expectedRoomNumber, $leastBed)
+    {
+        $roomQuantity = $this->roomQuantity->getById($id);
+        $bookingCardDetail = $this->bookingCardDetail->create([
+            'card_id' => $cardId,
+            'room_type_id' => $roomQuantity->roomType->id,
+            'quantity' => $quantity,
+
+        ]);
+
+        $newQuantity = $roomQuantity->vacancy_quantity - $quantity;
+
+        $roomQuantity->update([
+            'vacancy_quantity' => $newQuantity;
+
+        ]);
+    }
+
+    public function setBookingStatus($cardId, $currencyId)
+    {
+        $bookingCard = $this->bookingCard->getById($cardId);
+        $prices = $this->price->where('hotel_id', $bookingCard->hotel_id)->get();
+        $bill = 0;
+
+        foreach ($bookingCard->bookingCardDetails as $detail) {
+            foreach ($prices as $price) {
+                if ($detail->room_type_id == $price->room_type_id) {
+                    $pay = $price->amount * $detail->quantity;
+                    $value = $this->moneyExchange($price->currency, $pay, $currencyId);
+
+                    $bill += $value;
+                }
+            }
+        }
+
+        $bookingStatus = $this->bookingCardStatus->create([
+            'card_id' => $cardId,
+            'amount' => $bill,
+            'currency_id' => $currencyId,
+            'status_id' => config('default.init_status'),
+            
+        ]);
+    }
+
+    public function sendConfirmationMail($userId, $cardId)
+    {
+        $user = Auth::user();
+        $bookingCard = $this->bookingCard->getById($cardId);
+
+        Mail::send('email.booking',
+            [
+                'user' => $user,
+                'bookingCard' => $bookingCard,
+
+            ],
+            function ($message) use ($user) {
+                $message->to($user->email, $user->name)->subject(trans('booking.email_sub'));
+        });
+    }
+
+    public function booking(BookingRequest $request)
+    {
+        $checkIn = date(config('default.sql_date_format'), strtotime($request->get('checkin'));
+        $checkOut = date(config('default.sql_date_format'), strtotime($request->get('checkout'));
+        $code = substr(Auth::user()->name, 0, 3) . time();
+
+        $bookingCard = $this->bookingCard->create([
+            'user_id' => Auth::user()->id,
+            'card_code' => $code,
+            'hotel_id' => $request->get('hotel_id'),
+            'adult_quantity' => $request->get('adult-count'),
+            'children_quantity' => $request->get('children-count'),
+            'checkin_on' => $checkIn,
+            'checkout_on' => $checkOut,
+
+        ]);
+
+        $cardId = $bookingCard->id;
+
+        $hotel = $this->hotel->getById($request->get('hotel_id'));
+        foreach ($hotel->roomQuantities as $quantity) {
+            if (isset($request->get('rq-' . $quantity->id))) {
+                setBookingDetail($quantity->id, $request->get('rq-' . $quantity->id), $cardId));
+            }
+        }
```

Comment:
cho vào config

Reply:
à cái này không đổi đâu chị ơi. chỉ là cái input name không tương tác với DB đâu, em chỉ đăt để phân biệt khi mà số lượng tùy đối tượng sẽ in ra số input khác nhau thôi